src/ts/UIManager.ts

```diff
@@ -365,6 +374,7 @@ export class UIManager {
       this.managerPlayerWrapper.getPlayer().on(this.player.exports.PlayerEvent.SourceUnloaded, resolveUiVariant);
       this.managerPlayerWrapper.getPlayer().on(this.player.exports.PlayerEvent.Play, resolveUiVariant);
       this.managerPlayerWrapper.getPlayer().on(this.player.exports.PlayerEvent.Paused, resolveUiVariant);
+      this.managerPlayerWrapper.getPlayer().on(this.player.exports.PlayerEvent.Playing, resolveUiVariant);
```

Comment:
Didn't you want to resolve the UI variant (also) on `Ready`?

Reply:
Do you think Ready is needed? 🤔 We have the `SourceLoaded` already which I use as 'state' inside the Context. As I understand it, the duration (and therefore knowing if live or VOD) is available in the `SourceLoaded` event. The UI doesn't need to know if the Player is ready for immediate playback.

I only added the `PlayingEvent` here because the context already contains the `isPlaying` state, however the `resolveUiVariant` is not called in `Playing`.

Reply:
Understood and makes sense 🙂